applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -1580,6 +1599,30 @@ Vector UPwSmallStrainElement<TDim, TNumNodes>::CalculateGreenLagrangeStrain(cons
     return this->GetStressStatePolicy().CalculateGreenLagrangeStrain(rDeformationGradient);
 }
 
+template <unsigned int TDim, unsigned int TNumNodes>
+std::vector<double> UPwSmallStrainElement<TDim, TNumNodes>::CalculateDeterminantsOfDeformationGradients(
+    const std::vector<Matrix>& rDeformationGradients) const
+{
+    std::vector<double> result(rDeformationGradients.size());
+    std::transform(
+        rDeformationGradients.cbegin(), rDeformationGradients.cend(), result.begin(),
+        [](const auto& rDeformationGradient) { return MathUtils<>::Det(rDeformationGradient); });
+
+    return result;
+}
```

Comment:
Is it better to place this function in utilities?

Reply:
Moved this one to utilities, good point!